### PR TITLE
FIX: LNDhub URI invalid after BlueWallet update #7087

### PR DESCRIPTION
**Pull request by marcosrdz**
## Summary by CodeRabbit

- **New Features**
	- Introduced a new helper for managing LNDHub preferences, enhancing storage efficiency and error handling.
	- Added functions for retrieving, setting, and clearing LNDHub values.

- **Improvements**
	- Refactored logic for fetching and setting LNDHub URI to use a more robust storage method.
	- Enhanced address validation with improved normalization and error handling.
	- Improved code clarity and organization in the LightningSettings component.

- **Bug Fixes**
	- Corrected import path for `LightningSettings` to ensure consistency.

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

File: class/wallets/lightning-custodian-wallet.ts [hunk 1]
```diff
@@ -579,14 +579,17 @@ export class LightningCustodianWallet extends LegacyWallet {
     }
   }
 
-  static async isValidNodeAddress(address: string) {
-    const response = await fetch((address?.endsWith('/') ? address.slice(0, -1) : address) + '/getinfo', {
+  static async isValidNodeAddress(address: string): Promise<boolean> {
+    const normalizedAddress = new URL('/getinfo', address.replace(/([^:]\/)\/+/g, '$1'));
```
Comment (Overtorment) [inline class/wallets/lightning-custodian-wallet.ts:583]: hope this doesnt break it. cant parse it in my head, but looks suspicious

File: helpers/lndHub.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import DefaultPreference from 'react-native-default-preference';
+import { BlueApp } from '../class';
+import { GROUP_IO_BLUEWALLET } from '..//blue_modules/currency';
```
[the remark this answers is not in the document]
Comment (Overtorment) [reply helpers/lndHub.ts:4]: @marcosrdz

**Review by Overtorment — APPROVED** [review]

**Comment by GladosBlueWallet** [comment]
Unbelievable. You, [subject name here], must be the pride of [subject hometown here]!

**Pull request merged**